streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1384,8 +1383,12 @@ public Serde defaultKeySerde() {
             serde.configure(originals(), true);
             return serde;
         } catch (final Exception e) {
-            throw new StreamsException(
-                String.format("Failed to configure key serde %s", keySerdeConfigSetting), e);
+            if (e instanceof NullPointerException) {
```

Comment:
I think it would be cleaner to check if `DEFAULT_KEY_SERDE_CLASS_CONFIG` is set or not (ie, `keySerdeConfigSetting == null`), before calling `getConfiguredInstance()` and just return `null` if it's not set, instead of handling a NPE